lib/filterx/object-primitive.c

```diff
@@ -307,6 +308,10 @@ filterx_typecast_integer(FilterXExpr *s, GPtrArray *args)
         return filterx_integer_new(val);
     }
 
+  UnixTime ut;
+  if (filterx_object_extract_datetime(object, &ut))
+    return filterx_integer_new(ut.ut_sec * USEC_PER_SEC + ut.ut_usec);
```

Comment:
is this going to fit in a 64 bit integer?

Reply:
one ore question, I was thinking about increasing time resolution to nano seconds in the future, even in LogStamp, would that also fit? or we would lose some resolution in that case?

Reply:
Yes, it works with nanos, too for another ~240 years:

![image](https://github.com/user-attachments/assets/c3c43637-1f5f-4f12-933c-8caf5122e21e)